RealmSwift/EmbeddedObject.swift

```diff
@@ -217,6 +217,94 @@ extension EmbeddedObject: _RealmCollectionValueInsideOptional {
         return _observe(on: queue, block)
     }
 
+#if swift(>=5.8)
+    /**
+    Registers a block to be called each time the object changes.
+
+    The block will be asynchronously called on the given actor's executor after
+    each write transaction which deletes the object or modifies any of the managed
+    properties of the object, including self-assignments that set a property to its
+    existing value. The block is passed a copy of the object isolated to the
+    requested actor which can be safely used on that actor along with information
+    about what changed.
+
+    For write transactions performed on different threads or in different
+    processes, the block will be called when the managing Realm is
+    (auto)refreshed to a version including the changes, while for local write
+    transactions it will be called at some point in the future after the write
+    transaction is committed.
+
+    Only objects which are managed by a Realm can be observed in this way. You
+    must retain the returned token for as long as you want updates to be sent
+    to the block. To stop receiving updates, call `invalidate()` on the token.
+
+    By default, only direct changes to the object's properties will produce
+    notifications, and not changes to linked objects. Note that this is different
+    from collection change notifications. If a non-nil, non-empty keypath array is
+    passed in, only changes to the properties identified by those keypaths will
+    produce change notifications. The keypaths may traverse link properties to
+    receive information about changes to linked objects.
+
+    - warning: This method cannot be called during a write transaction, or when
+    the containing Realm is read-only.
+    - parameter actor: The actor to isolate notifications to.
+    - parameter block: The block to call with information about changes to the object.
+    - returns: A token which must be held for as long as you want updates to be delivered.
+     */
+    @available(macOS 10.15, tvOS 13.0, iOS 13.0, watchOS 6.0, *)
+    @_unsafeInheritExecutor
+    public func observe<A: Actor, T: Object>(
```

Comment:
Why can't the actor be assumed? Couldn't we pull the actor from the Realm?

Reply:
This function is specifically for observing on a different actor, similar to the option to observe on a different dispatch queue.